src/ophyd_async/epics/adcore/_core_detector.py

```diff
@@ -39,3 +41,40 @@ def get_plugin(
                 f"Expected {self.name}.{name} to be a {plugin_type}, got {plugin}"
             )
         return plugin
+
+
+class ContAcqAreaDetector(AreaDetector[ADBaseContAcqController]):
+    """Ophyd-async implementation of a continuously acquiring AreaDetector."""
+
+    def __init__(
+        self,
+        prefix: str,
+        path_provider: PathProvider,
+        drv_suffix: str = "cam1:",
```

Comment:
Should we add a `drv_cls = ADBaseIO` here in case we want to expose the extra PVs the class has? Most of the time I imagine that doesn't matter, but it would probably be useful to have the option...

Reply:
at the moment there is only one circular buffer class in areaDetector, so I think we should hardcode that, while there are many detector drivers that we should probably support

Reply:
Right, since the Circular Buffer plugin is a default plugin that will be included with every driver, and there is only one variant, I think we can leave as is. There are multiple different writers and drivers, hence passing in classes.

Reply:
Good with me then